lib/main.js

```diff
@@ -7,10 +6,9 @@ Thread.prototype.join = threx.join;
 Thread.prototype.enqueue = threx.enqueue;
 
 Thread.prototype.enqueue = function enqueue(obj, data) {
-  if (data instanceof Buffer &&
-      data.parent !== undefined)
+  if (data.parent !== undefined)
     throw new Error('Buffer should not be from a slice');
```

Comment:
This throw actually needs to be moved into C++, by using the `Buffer::HasInstance()` check. I'll take care of that in another patch.